chaindexing-postgres/src/repo.ts

```diff
@@ -47,16 +48,47 @@ export class PostgresRepo extends Repo<Pool, Conn> {
       });
   }
 
-  async streamContractAddresses(
-    conn: Conn,
-    streamer: (contractAddresses: ContractAddress[]) => void
-  ) {
-    streamer(await conn.query.chaindexingContractAddressesSchema.findMany());
-    // TODO: Implement serial streaming version
+  getContractAddressesStream(conn: Conn, opts?: { limit?: number }) {
+    let currentPage = 0;
+    let offset = 0;
+
+    const limit = opts?.limit || 10;
+
+    return {
+      next: async (): Promise<ContractAddress[] | null> => {
+        offset = limit * currentPage;
+        currentPage += 1;
+
+        return conn.query.chaindexingContractAddressesSchema.findMany({ limit, offset });
+      }
+    };
   }
 
-  async createEvents(_conn: Conn, _events: Event[]) {
-    // TODO: Implement
+  async createEvents(conn: Conn, events: Event[]) {
+    if (events.length === 0) return;
+
+    const _events = events.map((e) => ({ ...e, id: randomUUID() }));
+    await conn.insert(chaindexingEventsSchema).values(_events);
+  }
+
+  getEventsStream(conn: Conn, opts?: { limit?: number }) {
+    let currentPage = 0;
+    let offset = 0;
+
+    const limit = opts?.limit || 10;
+
+    return {
+      next: async (): Promise<Event[] | null> => {
+        offset = limit * currentPage;
+        currentPage += 1;
+
+        const result = conn.query.chaindexingEventsSchema.findMany({
+          limit,
+          offset
+        }) as unknown as Event[];
+        return result;
```

Comment:
The type should be inferred.

```suggestion
        return conn.query.chaindexingEventsSchema.findMany({
          limit,
          offset
        });
```